testament/categories.nim

```diff
@@ -475,47 +467,42 @@ proc getPackageDir(package: string): string =
   else:
     result = commandOutput[0].string
 
-iterator listPackages(filter: PackageFilter):
-                      tuple[name, url, cmd: string, hasDeps: bool] =
-  const defaultCmd = "nimble test"
+iterator listPackages(): tuple[name, url, cmd: string, hasDeps: bool] =
+  let defaultCmd = "nimble test"
   let packageList = parseFile(packageIndex)
-  for package in packageList.items:
-    if package.hasKey("url"):
+  for n, cmd, commit, hasDeps in important_packages.packages.items:
+    var found = false
+    for package in packageList.items:
       let name = package["name"].str
-      if name notin ["nimble", "compiler"]:
+      if name == n:
+        found = true
         let url = package["url"].str
-        case filter
-        of pfCoreOnly:
-          if "nim-lang" in normalize(url):
-            yield (name, url, defaultCmd, false)
-        of pfExtraOnly:
-          for n, cmd, commit, hasDeps in important_packages.packages.items:
-            if name == n:
-              let cmd = if cmd.len == 0: defaultCmd else: cmd
-              yield (name, url, cmd, hasDeps)
-        of pfAll:
-          yield (name, url, defaultCmd, false)
+        let cmd = if cmd.len == 0: defaultCmd else: cmd
+        yield (name, url, cmd, hasDeps)
+        break
```

Comment:
I think a ``yield + break`` turns this iterator into a regular proc?

Reply:
Not really.

`break` just takes you out of the inner `for` loop (where you search the whole nimble package list for your package `n` from the outer `for` loop).
After the `break`, you search for the next package.

Reply:
Ok.